antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -884,63 +895,37 @@ def exists(self, metadata: VariantStudy) -> bool:
     @override
     def copy(
         self,
-        src_meta: VariantStudy,
+        src_study: VariantStudy,
         dest_name: str,
         groups: Sequence[str],
         with_outputs: bool = False,
-    ) -> VariantStudy:
+    ) -> RawStudy:
         """
         Create a new variant study by copying a reference study.
 
         Args:
-            src_meta: The source study that you want to copy.
+            src_study: The source study that you want to copy.
             dest_name: The name for the destination study.
             groups: A list of groups to assign to the destination study.
             with_outputs: Indicates whether to copy the outputs as well.
 
         Returns:
             The newly created study.
         """
-        new_id = str(uuid4())
-        study_path = str(self.config.get_workspace_path() / new_id)
-        if src_meta.additional_data is None:
-            additional_data = StudyAdditionalData()
-        else:
-            additional_data = StudyAdditionalData(
-                horizon=src_meta.additional_data.horizon,
-                author=src_meta.additional_data.author,
-                patch=src_meta.additional_data.patch,
-            )
-        dst_meta = VariantStudy(
-            id=new_id,
-            name=dest_name,
-            parent_id=src_meta.parent_id,
-            path=study_path,
-            public_mode=PublicMode.NONE if groups else PublicMode.READ,
-            created_at=datetime.utcnow(),
-            updated_at=datetime.utcnow(),
-            version=src_meta.version,
-            groups=groups,
-            snapshot=None,
-            additional_data=additional_data,
-        )
 
-        # noinspection PyArgumentList
-        dst_meta.commands = [
-            CommandBlock(
-                study_id=new_id,
-                command=command.command,
-                args=command.args,
-                index=command.index,
-                version=command.version,
-                study_version=str(command.study_version),
-                user_id=command.user_id,
-                updated_at=command.updated_at,
-            )
-            for command in src_meta.commands
-        ]
+        self._safe_generation(src_study)
+
+        output_path = str(src_study.path + "/output")
+        src_study.path += "/snapshot"
```

Comment:
Actually, it's dangerous to modify the `Study` object this way: when we commit later, the object in database will be modified
--> we must avoid it

Again an option may be to provide the `FileStudy` to the raw service instead of the modified `VariantStudy`